stactask/task.py

```diff
@@ -108,15 +107,29 @@ def __init__(
 
     @property
     def process_definition(self) -> dict[str, Any]:
-        process = self._payload.get("process", {})
+        process = self._payload.get("process", [])
         if isinstance(process, dict):
+            warnings.warn(
+                "`process` is dictionary, use a list instead",
+                DeprecationWarning,
+                stacklevel=2,
+            )
             return process
-        else:
-            raise ValueError(f"process is not a dict: {type(process)}")
+
+        if not isinstance(process, list):
+            raise TypeError("`process` must be a list of dictionaries")
```

Comment:
Similarly here, workflow chaining in cirrus actually supports an array of arrays. That is, you could have something like this:

```
"process": [
    {"workflow": "A"},
    [{"workflow": "B1"}, {"workflow": "B2"}],
    [{"workflow": "C"}]
]
```

Where there's an array for a step the individual process definition objects will be unpacked into multiple payloads, making a branch in the process graph like this:

```
    +--> B1 --> C
A --+
    +--> B2 --> C
```

In other words, I think this message might be a bit misleading. What you know here is that if `process` is not a list, then it cannot be parsed, so maybe something in that vein is a better error message?

> unable to parse `process`: must be type list

Maybe?

Reply:
Yep. I like it. Done.